src/app/public/modules/datepicker/datepicker-input.directive.ts

```diff
@@ -420,6 +465,28 @@ export class SkyDatepickerInputDirective
     return dateValue;
   }
 
+  /**
+   * Validates the input value to ensure it is formatted correctly.
+   */
+  private isDateStringValid(value: string): boolean {
+    if (!value) {
+      return true;
+    }
+
+    // Does the value only include digits, dashes, or slashes?
+    const regexp = /^[\d\/\-]+$/;
+    const isValid = regexp.test(value);
+
+    if (isValid) {
+      return true;
+    }
+
+    // If not, does it conform to the standard ISO format?
+    const isValidIso = moment(value, moment.ISO_8601).isValid();
```

Comment:
Since you're using moment `isValid()` - wouldn't that cover all scenarios? Why do the regex check prior to this?

Reply:
This specific line is checking for ISO format, which is wildly different from `mm/dd/yyyy` (for example, `'2/12/2015'` would fail ISO format validation). A user will generally type in a user-friendly format (with slashes/dashes, etc.), but if an input is dynamically set to an ISO string (we support this currently), the input shouldn't fail, either.

Reply:
Got it. I didn't realize we supported ISO but did NOT support plain english dates (`March 3, 1980`). This makes sense now!